src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -755,93 +801,670 @@ public S(T tValue, ref T tRef, out T tOut, in T tIn)
         F = ref tRef;
         F = ref tOut;
         F = ref tIn;
-        F = tValue;
-        F = tRef;
-        F = tOut;
-        F = tIn;
     }
     object P
     {
         init
         {
-            F = ref GetValue();
             F = ref GetRef();
             F = ref GetRefReadonly();
-            F = GetValue();
-            F = GetRef();
-            F = GetRefReadonly();
         }
     }
-    static T GetValue() => throw null;
     static ref T GetRef() => throw null;
     static ref readonly T GetRefReadonly() => throw null;
+}";
+            // PROTOTYPE: Report a ref-safe-to-escape error for: F = ref tValue;
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition });
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void AssignValueTo_RefField()
+        {
+            var source =
+@"ref struct S<T>
+{
+    public ref T F;
+    public S(ref T t) { F = ref t; }
```

Comment:
no need to change it, but I wondered if the constructor affects anything here?

Reply:
No, the constructor has no effect in this test.